From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Fri, 31 Oct 2025 18:13:32 +0100
Subject: tray: Use one-shot idle/timeout functions

GLib added one-shot variants of the idle/timeout functions a while ago.

While they don't save a lot of code, they serve as a clear marker that
the callback is only invoked once.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/3947>
(cherry picked from commit cc1e00e8783beb443d50a5db1d59f5232604e2f5)
Origin: upstream, 49.4, commit:00f9f73b4a5289b1dcfbdc189f5f2b34effb3944
---
 src/tray/na-xembed.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/tray/na-xembed.c b/src/tray/na-xembed.c
index db51c41..5651768 100644
--- a/src/tray/na-xembed.c
+++ b/src/tray/na-xembed.c
@@ -253,7 +253,7 @@ na_xembed_synchronize_size (NaXembed *xembed)
     }
 }
 
-static gboolean
+static void
 synchronize_size_cb (gpointer user_data)
 {
   NaXembed *xembed = user_data;
@@ -261,8 +261,6 @@ synchronize_size_cb (gpointer user_data)
 
   na_xembed_synchronize_size (xembed);
   priv->resize_id = 0;
-
-  return G_SOURCE_REMOVE;
 }
 
 static void
@@ -299,7 +297,7 @@ na_xembed_resize (NaXembed *xembed)
   priv->have_size = TRUE;
   mtk_x11_error_trap_pop (xdisplay);
 
-  priv->resize_id = g_idle_add (synchronize_size_cb, xembed);
+  priv->resize_id = g_idle_add_once (synchronize_size_cb, xembed);
 }
 
 static gboolean
@@ -851,7 +849,7 @@ na_xembed_set_root_position (NaXembed *xembed,
   priv->root_y = y;
 
   if (priv->resize_id == 0)
-    priv->resize_id = g_idle_add (synchronize_size_cb, xembed);
+    priv->resize_id = g_idle_add_once (synchronize_size_cb, xembed);
 }
 
 MetaX11Display *
